### ISSUE TITLE
[perf] Unlocking Wallet hangs UI thread

### ISSUE THREAD

**Issue by atuchin-m** [brave/brave-browser#38280]
## Description 
Some Wallet UI task blocks UI thread for a long time during unlocking.

## Steps to Reproduce

   1. Create a Wallet if you don't have one;
   2. Start recording chrome://tracing 
   3. Open brave://wallet and unlock;
   4. Stop recording, check `BrowserMain` thread 

## Actual result:
A long UI task takes 254ms on MPB M1 pro.
On a low end android phone it probably takes a few seconds.

## Expected result:
No long UI tasks that blocks UI

## Reproduces how often: 
always

## Desktop Brave version:
1.68.2 Chromium: 125.0.6422.41 (Official Build) nightly (arm64)

**Comment by atuchin-m** [comment]
1. Unlocking a wallet could be a relatively usual user action (comparing to creating a new wallet).
2. 250ms is a really long time for M1 pro. For Android we probably have some ANRs because of that.

**Comment by atuchin-m** [comment]
cc @bbondy @supermassive

**Comment by supermassive** [comment]
yep, that's a known issue https://github.com/brave/brave-core/blob/v1.66.1/components/brave_wallet/browser/keyring_service.cc#L2447-L2450

**Comment by supermassive** [comment]
This https://github.com/brave/brave-core/pull/23363 should make unlocking 8-10 times faster, but still blocking. 
Will fix
